src/reject.rs

```diff
@@ -125,6 +126,51 @@ impl Rejection {
             .. self
         }
     }
+
+    /// Returns a json responder for this rejection.
+    pub fn json(&self) -> ::reply::Response {
+        use http::header::{CONTENT_TYPE, HeaderValue};
+        use hyper::Body;
+
+        let code = self.status();
+        let mut res = http::Response::default();
+        *res.status_mut() = code;
+
+        let empty = "{}".to_string();
```

Comment:
This will always allocate on the heap. I'd write it like this:

```rust
*res.body_mut() = match serde_json::to_string(&self) {
    Ok(s) => Body::from(s),
    Err(_) => Body::from("{}"),
};
```

Reply:
@seanmonstar 
> This will always allocate on the heap. I'd write it like this:

fixed